### ISSUE TITLE
Feature Request: Allow custom Relevant instances

### ISSUE THREAD

**Issue by kind-serge** [psibr/Contextually#1]
Currently, the static `AsyncLocal<InfoBlock>` is hidden behind static `Relevant` class. This works for a simple app. Once you want to build a more complex one, where multiple modules can use Contextually in different ways, this would become a problem due to absence of isolation of the contextual data of different modules.

My proposal is to make the `Relevant` class non-static, and allow custom instances of it. For example, I'm building a logging module that uses Contextually:

````csharp
static class Log
{
    private static Relevant _relevant = new Relevant();

    public static IDisposable Info(NameValueCollection info)
        => _relevant .Info(info);
}
````

```csharp
using (Log.Info(...))
{
  ...
}
```

**Comment by kind-serge** [comment]
Cool! But the `Dictionary<,>` is not thread-safe ;)

**Comment by aidapsibr** [comment]
Good catch, will switch to concurrentdictionary.

**Comment by aidapsibr** [comment]
@tyrotoxin Resolved in v1.2.1 #4
